api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/utils/resultsmatching/README.md

```diff
@@ -0,0 +1,59 @@
+# EPTS Reports Results Matcher
+> `ResultsMatchingTest.java` is an openmrs module sensitive context junit test that accesses an OpenMRS database and runs and matches results of an FGH report against its respective configured Jembi report in `eptsReportsResultsMatchingConfig.json`
+
+### Setup
+  - Disable @Ignore off `ResultsMatchingTest.java`
+  - Ensure your Mysql server with your openmrs database is running!
+  - Setup Mysql connection details in `~/.OpenMRS/openmrs-runtime.properties` or its equivalent, see sample below;
```

Comment:
I'm asking to see if there is a way to include these runtime properties within the test resources so that we don't need to navigate to a different place to configure these props...?

Reply:
For myself I usually have OpenMRS running in docker so there is no .OpenMRS folder locally. So to use this approach I need to load the module source code into the container and run the tests from there. I'm wondering if there is a way to work around this...?

Reply:
>Is it required to have OpenMRS installed for this test to work?

Not really, all you need to my knowledge is is mysql instance with your database being up and running.

>I'm asking to see if there is a way to include these runtime properties within the test resources so that we don't need to navigate to a different place to configure these props...?

I think yes, the openmrs module context sensitive test looks up the runtime properties file from [an openmrs folder configuration](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/util/OpenmrsUtil.java#L1024), setting OPENMRS_APPLICATION_DATA_DIRECTORY system variable should help with either `-D` on commandline or `System.setProperty`. the only challenge is we have varying credentials and may not make sense packaging them in a test jar

> For myself I usually have OpenMRS running in docker so there is no .OpenMRS folder locally. So to use this approach I need to load the module source code into the container and run the tests from there. I'm wondering if there is a way to work around this...?

You only need to be able to access your database through the host and through your connection url and credentials, your test can still access your database running in docker or in the openmrs instance in docker

Reply:
ok cool, thanks